### DISCUSSION TITLE
Ability to set charge limits

### DISCUSSION THREAD

**Discussion by aklnlh** [firstof9/openevse#218] [General]
Is there the ability to set charge limits based on time/energy/battery/range (as may be available to the evse).  
I know I can create an automation to monitor the kw charged for this session, or stop after a set time, but that relies on systems to communicate reliably.  (I've had communication issues in the past where it ended up charging the car to full, rather than the expected amount.)  If there was the ability to set the charge limits, this would also allow other systems to monitor what is going on with the charger, and 'plan' accordingly.

**Comment by firstof9** [comment]
> Is there the ability to set charge limits based on time/energy/battery/range (as may be available to the evse).

These are options available in OpenEVSE itself via the WebUI.

**Reply by aklnlh** [reply]
Yes, but was wondering if there was the ability in the integration to do it which I may have missed.

**Reply by firstof9** [reply]
Yes you can adjust the charge rate via the integration.

**Reply by wolterink** [reply]
I can see you can set the charge "RATE", but the question was if you can set a charge "LIMIT". It would be nice to set a charge "LIMIT" based on the solar forecast for the next day. Some days I want a full charge, but if a sunny day is forecasted there is no need for a full charge.

**Reply by firstof9** [reply]
Limits can currently be setup in the OpenEVSE UI itself.
I'll be adding this to the integration when I can.

**Reply by aklnlh** [reply]
Yes Limit was what I was querying about.  
If you need help in testing when you do get to the point of adding this to the integration, please let me know. Happy to help.
